gatorgrade/input/set_up_shell.py

```diff
@@ -0,0 +1,28 @@
+"""Set-up the shell commands."""
+import os, sys
+
+
+def run_setup(front_matter):
```

Comment:
I think this looks great overall! The only change I would recommend is maybe changing `front_matter` to something more clear like `setup_commands`. I just think `front_matter` could be a bit confusing, but I don't think it's necessary to change if others think it's fine.

Reply:
Actually I think `front_matter` is a better name here, since only the `setup` key inside `front_matter` actually contains the `setup_commands`. If other parameters to the first YAML document in `gatorgrade.yml` are added in the future, it would be even more accurate to call this dictionary `front_matter`.

Reply:
Alright, sounds good. It just confused me a little bit on initial impressions, but that makes a lot more sense.